lib/Serialization/Deserialization.cpp

```diff
@@ -1256,6 +1257,13 @@ findNestedTypeDeclInModule(FileUnit *thisFile, ModuleDecl *extensionModule,
   return nullptr;
 }
 
+/// Look for nested types in all files of \p extensionModule.
+static TypeDecl *
+findNestedTypeDeclInModule(ModuleDecl *extensionModule,
+                           Identifier name, NominalTypeDecl *parent)  {
+  return findNestedTypeDeclInModule(nullptr, extensionModule, name, parent);
```

Comment:
This makes me nervous since we can now set up a cross-ref that causes resolution to re-enter itself - especially through a name lookup. This was the cause of a ton of circular lookup problems in the SDK when we were refactoring the clang importer. Lazy member loading no longer has re-entrancy guards in it to catch this too...

Reply:
Wouldn't a circular issue be detected in the frontend invocations before reaching merge-modules?
But regardless, it looks to me that this particular code-path follows the strict `xref` path components, so it's going to reach the end of the lookup by definition.